### ISSUE TITLE
Including Wire breaks FR2433

### ISSUE THREAD

**Issue by MB3hel** [MB3hel/Arduino-Core-MSP430#1]
Minimal example:
```cpp
#include <Wire.h>

void setup(){
  pinMode(P1_0, OUTPUT);
  return;
}

void loop(){
  digitalWrite(P1_0, !digitalRead(P1_0));
  delay(500);
  return;
}
```

- Compile for FR2433 **without link time optimizations**. The LED does not blink.
- Compile with link time optimizations. The LED does blink.

Does not seem to impact other boards (at least not ones I have).

- Works in both cases on FR2355
- Works in both cases on FR5969
- Works in both cases on G2553

**Comment by MB3hel** [comment]
Debugging notes:
- Removing contents of wire.cpp does not fix
- Removing contents of twi_sw.c does not fix
- Removing contents of twi.c does fix
- Adding contents of twi_sw.c back does not break

Issue seems to be related to how the linker / compiler handles something in twi.c

- Removing eusci_isr_handler fixes the issue
- Removing call to eusci_isr_handler in USCIB0_ISR fixes the issue
- replacing the contents of eusci_isr_handler with `return 0` fixes the issue
- Always returning 0 from eusci_isr_handler at the end does **not** fix the issue

**Comment by MB3hel** [comment]
More Debug Notes:
- This seems to occur if `eusci_isr_handler` is linked into the program
- This can be forced in a way such that it never actually gets called (see attached zip) and such that it is not called via ISR
- ISR is still ensured to be in the program (by calling `usci_isr_install` and twi functions forced to link with `twi_dummy`)

This program fails to work as intended (for some reason)

If the contents of the function are replaced with a nop things work correctly (note that the function is forced to not be inlined for consistency)

[Arduino-Core-MSP430-temp-breakwire.zip](https://github.com/MB3hel/Arduino-Core-MSP430/files/10833040/Arduino-Core-MSP430-temp-breakwire.zip)

**Comment by MB3hel** [comment]
Related to the value assigned to I2C_baseAddress being 0x0540 which seems to break things (suspected compiler bug). Workaround pushed that should avoid this. Leaving open until either confirmed compiler bug or other solution is found.

**Comment by MB3hel** [comment]
More investigation shows that this breaks things when bss is not empty and the first thing in data is 0x0540 (or various other 0x_40 values). Suspected cause is errata CPU-40

**Comment by MB3hel** [comment]
Fixed by 9e7e191144afb2e3911dc58dfb1e8c6b95a42441 in modified linker scripts
